clouddrift/analysis.py

```diff
@@ -81,14 +82,16 @@ def apply_ragged(
     arrays = [unpack_ragged(arr, rowsize) for arr in arrays]
     iter = [[arrays[i][j] for i in range(len(arrays))] for j in range(len(arrays[0]))]
 
-    # combine other arguments
-    for arg in iter:
-        if args:
-            arg.append(*args)
+    # threads or processes
+    if use_threads:
+        executor_type = futures.ThreadPoolExecutor
+    else:
+        executor_type = futures.ProcessPoolExecutor
 
     # parallel execution
-    with futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
-        res = executor.map(lambda x: func(*x, **kwargs), iter)
+    with executor_type(max_workers=max_workers) as executor:
+        res = [executor.submit(func, *x, *args, **kwargs).result() for x in iter]
```

Comment:
I also switch to submit, which in my opinion, simplifies the syntax since we have `*args` and `**kwargs` to pass to the function.